backend/middleware/documents.js

```diff
@@ -0,0 +1,43 @@
+import { User } from "../models/userModel.js";
+import FileObject from "../models/fileModel.js";
+
+/*
+authenticateUser() from auth.js must be used before this
+*/
+export const authenticateDocumentPermissions = async (req, res, next) => {
+    try{
+        const email = req.user.email;
+        const user = await User.findOne({ email }).exec();
+        const documentId = req.params.id;
+        const document = await FileObject.findOne({_id: documentId}).populate("uploader")
+        switch(user.role){
+            case "veteran":
+                if(!document.uploader.equals(user)){
+                    return res.status(500).json({ error: "You do not have permission to view this document!" });
```

Comment:
Not a huge deal but in all these object-level permissions middlewares, it's better to return HTTP 403 (Forbidden) instead of HTTP 500 (Internal Server Error) so the frontend can display different error messages - 403 means the user doesn't have permission to access the resource, while 500 means something went wrong and we don't know what